contracts/NodeDelegator.sol

```diff
@@ -206,15 +218,29 @@ contract NodeDelegator is INodeDelegator, LRTConfigRoleChecker, PausableUpgradea
     function getAssetBalance(address asset) public view override returns (uint256 ndcAssets, uint256 eigenAssets) {
         ndcAssets += IERC20(asset).balanceOf(address(this));
 
+        // The WETH asset will point to the EigenLayer beaconChainETHStrategy 0xbeaC0eeEeeeeEEeEeEEEEeeEEeEeeeEeeEEBEaC0
+        address strategy = lrtConfig.assetStrategy(asset);
+        if (strategy != address(0)) {
+            // Get the amount of strategy shares owned by this NodeDelegator contract
+            IStrategyManager strategyManager =
+                IStrategyManager(lrtConfig.getContract(LRTConstants.EIGEN_STRATEGY_MANAGER));
+            uint256 strategyShares = strategyManager.stakerStrategyShares(address(this), IStrategy(strategy));
+
+            // add any shares pending withdrawal to the strategy share owned by this NodeDelegator
+            strategyShares += pendingInternalShareWithdrawals[strategy];
```

Comment:
Does this do exactly the wrong thing?

We calculate the worth of primeETH tokens for minting/burning/etc by computing prime assets vs prime totalSupply.

When a withdraw starts, the prime tokens are burned. If the reported assets stayed the same, this would make the tokens appear to have more value, which would allow a subsequent withdraw to withdraw more prime eth for the same number of prime tokens, and repeat.

I think we want to ensure that as we burn prime eth tokens, we exactly match how much we remove from our asset accounting.

Reply:
I'll update the comment so it's clear this is only for internal withdrawals by the Prime operator. Staker withdrawals do not add to `pendingInternalShareWithdrawals` as the stakers tokens have been burned. The problem is only with internal withdrawals where primeETH is not burned but the underlying assets will disappear until the claim.